sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/ResourceGroupOperations.cs

```diff
@@ -111,7 +142,7 @@ public virtual async Task<Response> DeleteAsync(CancellationToken cancellationTo
 
             try
             {
-                return await Operations.StartDelete(Id.Name, cancellationToken).WaitForCompletionAsync(cancellationToken).ConfigureAwait(false);
+                return await RestClient.DeleteAsync(Id.Name, cancellationToken).ConfigureAwait(false);
```

Comment:
the deleteAsync method should call the StartDeleteAsync method then WaitForCompletionAsync